### PR TITLE
Reworked Aftershock's UPS to custom battery

### PR DESCRIPTION
**Pull request by gwyter**
#### Summary
```SUMMARY: None```

#### Purpose of change

Make it impossible to recharge disposable batteries through an UPS exploit, as stated in #37315 .
#### Describe the solution

Removed the RECHARGE flag from mods/Aftershock/items/tools.json, this way the player can't drop an UPS loaded with a disposable battery and have it recharge in a recharging station.

#### Describe alternatives you've considered

Changes directly to the UPS code, maybe in dda's or Aftershock's, but i don't think it'd be as simple as this solution, and i wouldn't know how to work around with c++ yet.

#### Testing

I tested debug spawning a vehicle and installing a recharging station, then spawned an UPS, loaded it with a heavy disposable battery, drained some of the charge with a few laser shots, dropped the heavy disposable battery-loaded UPS in the recharging station making sure it was turned on, and the UPS didn't recharge the battery, as expected.

#### Additional context

The downside to this fix is the extra key presses, as the player will have to unload the UPS to recharge it's battery and then reload it after charged, but that's not much more of an effort, and it seems simpler than working out a deeper solution involving messing with the recharging code, or the UPS specifically.

### CODE REVIEW

**Comment by John-Candlebury** [comment]
I'd prefer if you make aftershock's  UPS have a high density integral battery, that can be recharged normally, but that cant be removed from it.

**Comment by Maleclypse** [comment]
I get confused on flags so I wanted to check does the `NO_RELOAD` flag make the UPS itself non-reloadable?

**Comment by gwyter** [comment]
@Maleclypse yes, it means you can't put a magazine on it, nor pure "battery" charges.

File: data/mods/Aftershock/items/tools.json [hunk 1]
```diff
@@ -217,11 +217,15 @@
...
+    "description": "This is a unified power supply, or UPS.  It is a device developed jointly by military and scientific interests for use in combat and the field.  The UPS is designed to power armor and some guns, but drains batteries quickly.  It can be worn around to either leg for ease of access, and it's been waterproofed to protect the delicate electronics. Has it's own custom battery, with higher capacity and rechargeable, but not removeable",
     "coverage": 5,
+    "ammo": "battery",
+    "initial_charges": 1000,
+    "max_charges": 1500,
     "encumbrance": 2,
     "covers": [ "LEG_EITHER" ],
-    "flags": [ "RECHARGE", "WAIST", "FRAGILE", "OVERSIZE", "WATERPROOF", "IS_UPS" ]
+    "flags": [ "RECHARGE", "WAIST", "FRAGILE", "OVERSIZE", "WATERPROOF", "IS_UPS", "NO_RELOAD", "NO_UNLOAD" ],
+    "magazines": []
```
Comment (John-Candlebury) [inline data/mods/Aftershock/items/tools.json]:
```suggestion
    "magazines": [  ]
```

**Comment by Unrepentant-Atheist** [comment]
The point of the UPS then being...?

**Comment by gwyter** [comment]
@Unrepentant-Atheist I think it's an overall improvement even if you think solely about usefulness.

1. It has more charges than most other batteries, thus removing the need to get a heavy high capacity battery for it to be more usable.
1. It's simpler not to worry about specific batteries, you just drop the UPS in a recharging station and pick it up again after it's done. 
1. It makes it easier for a player with resources to just make multiple UPS's and walk around like a power tank, not forgetting they're still fragile pieces of equipment, if that's your definition of useful, as you won't have to micro-manage gazillion of batteries, as the 2500s ones are the car batteries, which are obscenely heavy for a human to carry in quantities, and the heavy disposable batteries, which aren't supposed to be rechargeable at all.

That said, I don't think it'd be unreasonable to propose something like an "UPSpack", a cluster of other UPS adapted for bigger storage, but that's always proposable as a request maybe by you or someone else.

**Comment by Unrepentant-Atheist** [comment]
You're telling me, it makes more sense, to unpack UPS, drop batteries into charger, recharge, and pack UPS with recharged batteries instead of....attaching a depleted UPS with packed batteries, where the UPS have cables and ports hanging out into a recharger?

Are you seriously telling me that? So, you take out the batteries from you cell phone to recharge them? Thankfully this game is open source, so I can circumvent idiotic PRs like this.

**Comment by gwyter** [comment]
@Unrepentant-Atheist I think you've read it wrong, the UPS is exactly that now, it's an item which by itself has 1500 maximum charge capacity, so you just drop it into a recharging station and it'll recharge, what has happened is that there's no more a need to craft/find/load a battery into the UPS, it already has it's own non-removeable and rechargeable battery.

But i see your confusion as the title hasn't been changed since the PR has started, overall i'd tell you to try the change out, as the just dropping depleted UPS in recharger and charge them is not a removed feature.

**Comment by Unrepentant-Atheist** [comment]
Guess I have to re-craft the UPS then, because mine with 1500 batteries can't be recharged.

**Pull request merged**